File: src/interaction/Zero.cpp
/*
  Copyright (C) 2012,2013
      Max Planck Institute for Polymer Research
  Copyright (C) 2008,2009,2010,2011
      Max-Planck-Institute for Polymer Research & Fraunhofer SCAI
  
  This file is part of ESPResSo++.
  
  ESPResSo++ is free software: you can redistribute it and/or modify
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation, either version 3 of the License, or
  (at your option) any later version.
  
  ESPResSo++ is distributed in the hope that it will be useful,
  but WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  GNU General Public License for more details.
  
  You should have received a copy of the GNU General Public License
  along with this program.  If not, see <http://www.gnu.org/licenses/>. 
*/

#include "python.hpp"
#include "Zero.hpp"
#include "Tabulated.hpp"
#include "VerletListInteractionTemplate.hpp"
#include "VerletListAdressInteractionTemplate.hpp"
#include "VerletListHadressInteractionTemplate.hpp"
#include "CellListAllPairsInteractionTemplate.hpp"
#include "FixedPairListInteractionTemplate.hpp"

namespace espressopp {
  namespace interaction {
      
    typedef class VerletListInteractionTemplate <Zero>
        VerletListZero;
    typedef class VerletListAdressInteractionTemplate <Zero, Tabulated>
        VerletListAdressZero;
    typedef class VerletListHadressInteractionTemplate <Zero, Tabulated>
        VerletListHadressZero;
    typedef class CellListAllPairsInteractionTemplate <Zero>
        CellListZero;
    typedef class FixedPairListInteractionTemplate <Zero>
        FixedPairListZero;

    //////////////////////////////////////////////////
    // REGISTRATION WITH PYTHON
    //////////////////////////////////////////////////
    void 
    Zero::registerPython() {
      using namespace espressopp::python;

      class_< Zero, bases< Potential > >
    	("interaction_Zero", init<>())
	    .def(init<>())
	    .def_pickle(Zero_pickle())
      ;

      class_< VerletListZero, bases< Interaction > >
        ("interaction_VerletListZero", init< shared_ptr<VerletList> >())
        .def("setPotential", &VerletListZero::setPotential, return_value_policy< reference_existing_object >())
        .def("getPotential", &VerletListZero::getPotential, return_value_policy< reference_existing_object >())
      ;

      class_< VerletListAdressZero, bases< Interaction > >
        ("interaction_VerletListAdressZero",
           init< shared_ptr<VerletListAdress>,
                  shared_ptr<FixedTupleListAdress> >())
        .def("setFixedTupleList", &VerletListAdressZero::setFixedTupleList)
        .def("setPotentialAT", &VerletListAdressZero::setPotentialAT)
        .def("setPotentialCG", &VerletListAdressZero::setPotentialCG);
      ;

      class_< VerletListHadressZero, bases< Interaction > >
        ("interaction_VerletListHadressZero",
           init< shared_ptr<VerletListAdress>,
                  shared_ptr<FixedTupleListAdress> >())
        .def("setFixedTupleList", &VerletListHadressZero::setFixedTupleList)
        .def("setPotentialAT", &VerletListHadressZero::setPotentialAT)
        .def("setPotentialCG", &VerletListHadressZero::setPotentialCG);
      ;
      
      class_< CellListZero, bases< Interaction > >
        ("interaction_CellListZero", init< shared_ptr< storage::Storage > >())
        .def("setPotential", &CellListZero::setPotential);
	  ;

      class_< FixedPairListZero, bases< Interaction > >
        ("interaction_FixedPairListZero",
          init< shared_ptr<System>, shared_ptr<FixedPairList>, shared_ptr<Zero> >())
          .def("setPotential", &FixedPairListZero::setPotential);
      ;
    }
    
  }
}

File: src/interaction/Zero.py
#  Copyright (C) 2012,2013
#      Max Planck Institute for Polymer Research
#  Copyright (C) 2008,2009,2010,2011
#      Max-Planck-Institute for Polymer Research & Fraunhofer SCAI
#  
#  This file is part of ESPResSo++.
#  
#  ESPResSo++ is free software: you can redistribute it and/or modify
#  it under the terms of the GNU General Public License as published by
#  the Free Software Foundation, either version 3 of the License, or
#  (at your option) any later version.
#  
#  ESPResSo++ is distributed in the hope that it will be useful,
#  but WITHOUT ANY WARRANTY; without even the implied warranty of
#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#  GNU General Public License for more details.
#  
#  You should have received a copy of the GNU General Public License
#  along with this program.  If not, see <http://www.gnu.org/licenses/>. 


r"""
***************************
espressopp.interaction.Zero
***************************

This class provides methods for a zero potential
no interactions between particles, mainly used for debugging and testing





.. function:: espressopp.interaction.Zero()


.. function:: espressopp.interaction.VerletListZero(vl)

		:param vl: 
		:type vl: 

.. function:: espressopp.interaction.VerletListZero.getPotential(type1, type2)

		:param type1: 
		:param type2: 
		:type type1: 
		:type type2: 
		:rtype: 

.. function:: espressopp.interaction.VerletListZero.setFixedTupleList(ftpl)

		:param ftpl: 
		:type ftpl: 

.. function:: espressopp.interaction.VerletListZero.setPotential(type1, type2, potential)

		:param type1: 
		:param type2: 
		:param potential: 
		:type type1: 
		:type type2: 
		:type potential: 

.. function:: espressopp.interaction.VerletListAdressZero(vl)

		:param vl: 
		:type vl: 

.. function:: espressopp.interaction.VerletListAdressZero.setFixedTupleList(ftpl)

		:param ftpl: 
		:type ftpl: 

.. function:: espressopp.interaction.VerletListAdressZero.setPotentialAT(type1, type2, potential)

		:param type1: 
		:param type2: 
		:param potential: 
		:type type1: 
		:type type2: 
		:type potential: 

.. function:: espressopp.interaction.VerletListAdressZero.setPotentialCG(type1, type2, potential)

		:param type1: 
		:param type2: 
		:param potential: 
		:type type1: 
		:type type2: 
		:type potential: 

.. function:: espressopp.interaction.VerletListHadressZero(vl, fixedtupleList)

		:param vl: 
		:param fixedtupleList: 
		:type vl: 
		:type fixedtupleList: 

.. function:: espressopp.interaction.VerletListHadressZero.setFixedTupleList(ftpl)

		:param ftpl: 
		:type ftpl: 

.. function:: espressopp.interaction.VerletListHadressZero.setPotentialAT(type1, type2, potential)

		:param type1: 
		:param type2: 
		:param potential: 
		:type type1: 
		:type type2: 
		:type potential: 

.. function:: espressopp.interaction.VerletListHadressZero.setPotentialCG(type1, type2, potential)

		:param type1: 
		:param type2: 
		:param potential: 
		:type type1: 
		:type type2: 
		:type potential: 

.. function:: espressopp.interaction.CellListZero(stor)

		:param stor: 
		:type stor: 

.. function:: espressopp.interaction.CellListZero.setPotential(type1, type2, potential)

		:param type1: 
		:param type2: 
		:param potential: 
		:type type1: 
		:type type2: 
		:type potential: 

.. function:: espressopp.interaction.FixedPairListZero(system, vl, potential)

		:param system: 
		:param vl: 
		:param potential: 
		:type system: 
		:type vl: 
		:type potential: 

.. function:: espressopp.interaction.FixedPairListZero.setPotential(potential)

		:param potential: 
		:type potential: 
"""
from espressopp import pmi
from espressopp.esutil import *

from espressopp.interaction.Potential import *
from espressopp.interaction.Interaction import *
from _espressopp import interaction_Zero, \
                      interaction_VerletListZero, \
                      interaction_VerletListAdressZero, \
                      interaction_VerletListHadressZero, \
                      interaction_CellListZero, \
                      interaction_FixedPairListZero

class ZeroLocal(PotentialLocal, interaction_Zero):
    def __init__(self):

        if not (pmi._PMIComm and pmi._PMIComm.isActive()) or pmi._MPIcomm.rank in pmi._PMIComm.getMPIcpugroup():
            cxxinit(self, interaction_Zero)

class VerletListZeroLocal(InteractionLocal, interaction_VerletListZero):
    def __init__(self, vl):
        if not (pmi._PMIComm and pmi._PMIComm.isActive()) or pmi._MPIcomm.rank in pmi._PMIComm.getMPIcpugroup():
            cxxinit(self, interaction_VerletListZero, vl)

    def setPotential(self, type1, type2, potential):
        if not (pmi._PMIComm and pmi._PMIComm.isActive()) or pmi._MPIcomm.rank in pmi._PMIComm.getMPIcpugroup():
            self.cxxclass.setPotential(self, type1, type2, potential)
            
    def getPotential(self, type1, type2):
        if not (pmi._PMIComm and pmi._PMIComm.isActive()) or pmi._MPIcomm.rank in pmi._PMIComm.getMPIcpugroup():
            return self.cxxclass.getPotential(self, type1, type2)
            
    def setFixedTupleList(self, ftpl):
        if not (pmi._PMIComm and pmi._PMIComm.isActive()) or pmi._MPIcomm.rank in pmi._PMIComm.getMPIcpugroup():
            self.cxxclass.setFixedTupleList(self, ftpl)

class VerletListAdressZeroLocal(InteractionLocal, interaction_VerletListAdressZero):
    def __init__(self, vl):
        if not (pmi._PMIComm and pmi._PMIComm.isActive()) or pmi._MPIcomm.rank in pmi._PMIComm.getMPIcpugroup():
            cxxinit(self, interaction_VerletListAdressZero, vl)

    def setPotentialAT(self, type1, type2, potential):
        if not (pmi._PMIComm and pmi._PMIComm.isActive()) or pmi._MPIcomm.rank in pmi._PMIComm.getMPIcpugroup():
            self.cxxclass.setPotentialAT(self, type1, type2, potential)
            
    def setPotentialCG(self, type1, type2, potential):
        if not (pmi._PMIComm and pmi._PMIComm.isActive()) or pmi._MPIcomm.rank in pmi._PMIComm.getMPIcpugroup():
            self.cxxclass.setPotentialAT(self, type1, type2, potential)            
            
    def setFixedTupleList(self, ftpl):
        if not (pmi._PMIComm and pmi._PMIComm.isActive()) or pmi._MPIcomm.rank in pmi._PMIComm.getMPIcpugroup():
            self.cxxclass.setFixedTupleList(self, ftpl)
            
class VerletListHadressZeroLocal(InteractionLocal, interaction_VerletListHadressZero):
    def __init__(self, vl, fixedtupleList):
        if not (pmi._PMIComm and pmi._PMIComm.isActive()) or pmi._MPIcomm.rank in pmi._PMIComm.getMPIcpugroup():
            cxxinit(self, interaction_VerletListHadressZero, vl, fixedtupleList)

    def setPotentialAT(self, type1, type2, potential):
        if not (pmi._PMIComm and pmi._PMIComm.isActive()) or pmi._MPIcomm.rank in pmi._PMIComm.getMPIcpugroup():
            self.cxxclass.setPotentialAT(self, type1, type2, potential)
            
    def setPotentialCG(self, type1, type2, potential):
        if not (pmi._PMIComm and pmi._PMIComm.isActive()) or pmi._MPIcomm.rank in pmi._PMIComm.getMPIcpugroup():
            self.cxxclass.setPotentialAT(self, type1, type2, potential)            
            
    def setFixedTupleList(self, ftpl):
        if not (pmi._PMIComm and pmi._PMIComm.isActive()) or pmi._MPIcomm.rank in pmi._PMIComm.getMPIcpugroup():
            self.cxxclass.setFixedTupleList(self, ftpl)

class CellListZeroLocal(InteractionLocal, interaction_CellListZero):
    def __init__(self, stor):
        if not (pmi._PMIComm and pmi._PMIComm.isActive()) or pmi._MPIcomm.rank in pmi._PMIComm.getMPIcpugroup():
            cxxinit(self, interaction_CellListZero, stor)
        
    def setPotential(self, type1, type2, potential):
        if not (pmi._PMIComm and pmi._PMIComm.isActive()) or pmi._MPIcomm.rank in pmi._PMIComm.getMPIcpugroup():
            self.cxxclass.setPotential(self, type1, type2, potential)

class FixedPairListZeroLocal(InteractionLocal, interaction_FixedPairListZero):
    def __init__(self, system, vl, potential):
        if not (pmi._PMIComm and pmi._PMIComm.isActive()) or pmi._MPIcomm.rank in pmi._PMIComm.getMPIcpugroup():
            cxxinit(self, interaction_FixedPairListZero, system, vl, potential)
        
    def setPotential(self, potential):
        if not (pmi._PMIComm and pmi._PMIComm.isActive()) or pmi._MPIcomm.rank in pmi._PMIComm.getMPIcpugroup():
            self.cxxclass.setPotential(self, potential)

if pmi.isController:
    class Zero(Potential):
        'The Zero potential.'
        pmiproxydefs = dict(
            cls = 'espressopp.interaction.ZeroLocal'
            )

    class VerletListZero(Interaction):
        __metaclass__ = pmi.Proxy
        pmiproxydefs = dict(
            cls =  'espressopp.interaction.VerletListZeroLocal',
            pmicall = ['setPotential', 'getPotential', 'setFixedTupleList']
            )

    class VerletListAdressZero(Interaction):
        __metaclass__ = pmi.Proxy
        pmiproxydefs = dict(
            cls =  'espressopp.interaction.VerletListAdressZeroLocal',
            pmicall = ['setPotentialAT', 'setPotentialCG']
            )
            
    class VerletListHadressZero(Interaction):
        __metaclass__ = pmi.Proxy
        pmiproxydefs = dict(
            cls =  'espressopp.interaction.VerletListHadressZeroLocal',
            pmicall = ['setPotentialAT', 'setPotentialCG']
            )

    class CellListZero(Interaction):
        __metaclass__ = pmi.Proxy
        pmiproxydefs = dict(
            cls =  'espressopp.interaction.CellListZeroLocal',
            pmicall = ['setPotential']
            )
        
    class FixedPairListZero(Interaction):
        __metaclass__ = pmi.Proxy
        pmiproxydefs = dict(
            cls =  'espressopp.interaction.FixedPairListZeroLocal',
            pmicall = ['setPotential']
            )
